aws/resource_aws_codedeploy_deployment_group.go

```diff
@@ -538,6 +712,98 @@ func buildAlarmConfig(configured []interface{}) *codedeploy.AlarmConfiguration {
 	return result
 }
 
+// buildDeploymentStyle converts a raw schema list containing a map[string]interface{}
+// into a single codedeploy.DeploymentStyle object
+func buildDeploymentStyle(list []interface{}) *codedeploy.DeploymentStyle {
+	result := &codedeploy.DeploymentStyle{}
+
+	if len(list) == 1 {
+		style := list[0].(map[string]interface{})
+		result.DeploymentOption = aws.String(style["deployment_option"].(string))
+		result.DeploymentType = aws.String(style["deployment_type"].(string))
+	}
+
+	return result
+}
+
+// buildLoadBalancerInfo converts a raw schema list containing a map[string]interface{}
+// into a single codedeploy.LoadBalancerInfo object
+func buildLoadBalancerInfo(list []interface{}) *codedeploy.LoadBalancerInfo {
+	if len(list) == 0 || list[0] == nil {
+		return nil
+	}
+
+	loadBalancerInfo := &codedeploy.LoadBalancerInfo{}
+
+	lbInfo := list[0].(map[string]interface{})
+
+	if attr, ok := lbInfo["elb_info"]; ok {
+		elbs := attr.(*schema.Set).List()
+		loadBalancerInfo.ElbInfoList = make([]*codedeploy.ELBInfo, 0, len(elbs))
+
+		for _, v := range elbs {
+			elb := v.(map[string]interface{})
+			loadBalancerInfo.ElbInfoList = append(loadBalancerInfo.ElbInfoList, &codedeploy.ELBInfo{
+				Name: aws.String(elb["name"].(string)),
+			})
+		}
+	}
+
+	if attr, ok := lbInfo["target_group_info"]; ok {
+		targetGroups := attr.(*schema.Set).List()
+		loadBalancerInfo.TargetGroupInfoList = make([]*codedeploy.TargetGroupInfo, 0, len(targetGroups))
+
+		for _, v := range targetGroups {
+			targetGroup := v.(map[string]interface{})
+			loadBalancerInfo.TargetGroupInfoList = append(loadBalancerInfo.TargetGroupInfoList, &codedeploy.TargetGroupInfo{
+				Name: aws.String(targetGroup["name"].(string)),
+			})
+		}
+	}
+
+	return loadBalancerInfo
+}
+
+// buildBlueGreenDeploymentConfig converts a raw schema list containing a map[string]interface{}
+// into a single codedeploy.BlueGreenDeploymentConfiguration object
+func buildBlueGreenDeploymentConfig(list []interface{}) *codedeploy.BlueGreenDeploymentConfiguration {
+	if len(list) == 0 || list[0] == nil {
+		return nil
+	}
+
+	blueGreenDeploymentConfig := &codedeploy.BlueGreenDeploymentConfiguration{}
+
+	config := list[0].(map[string]interface{})
+
+	if attr, ok := config["deployment_ready_option"]; ok {
+		m := attr.([]interface{})[0].(map[string]interface{})
+		deploymentReadyOption := &codedeploy.DeploymentReadyOption{
+			ActionOnTimeout:   aws.String(m["action_on_timeout"].(string)),
+			WaitTimeInMinutes: aws.Int64(int64(m["wait_time_in_minutes"].(int))),
```

Comment:
As mentioned above - these are also optional fields, so we should only be passing them to the API if they're actually defined in the config.